nvidia_fan_controller.py

```diff
@@ -194,22 +194,37 @@ def run_cmd(cmd: List[str]) -> str:
     return p.stdout.read().decode()
 
 
+def get_fans() -> Dict[int, List[int]]:
+    stdout = run_cmd(['nvidia-settings', '-q', 'gpus', '--verbose'])
+
+    gpu_fan_info = {}
+    gpu_blocks = re.findall(r'\[gpu:(\d+)\](.*?)Sensor \d+\)', stdout, flags=re.DOTALL)
+    for gpu_id, gpu_block in gpu_blocks:
+        fan_ids = re.findall(r'\[fan:(\d+)\]', gpu_block)
+        gpu_fan_info[int(gpu_id)] = [int(fan_id) for fan_id in fan_ids]
+
+    for gpu_id in gpu_fan_info:
```

Comment:
Do we need this second for loop?

Reply:
Well, my regex is rusty indeed. I could imagine this to be working in one loop, but this section is not performance critical, it only gets called once.